projects/ADuCM360_demo_cn0411/src/CN0411.c

```diff
@@ -0,0 +1,1876 @@
+ * CN0411 ADC Channel 5 configuration function
+ *
+ * @param cn0411_dev - The device structure.
+ * @param ch_en - enable/disable ADC channel
+ * @return 0 in case of success, negative error code otherwise.
+*/
+
+int32_t CN0411_ADC_ch5 (cn0411_device *cn0411_dev, uint8_t ch_en)
+{
+	int32_t ret;
+
+	cn0411_dev->ad7124_dev.regs[AD7124_Channel_5].value =
+		((ch_en << 15)
+		 | AD7124_CH_MAP_REG_SETUP(1)
+		 | AD7124_CH_MAP_REG_AINP(11)
+		 | AD7124_CH_MAP_REG_AINM(17));
+	ret = AD7124_WriteRegister(&cn0411_dev->ad7124_dev,
+				   cn0411_dev->ad7124_dev.regs[AD7124_Channel_5]);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC IO Control 1 configuration function
+ *
+ * @param cn0411_dev - The device structure.
+ * @param ch_gain - select gain channel to be used.
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_set_io1 (cn0411_device *cn0411_dev, uint8_t ch_gain)
+{
+	int32_t ret;
+
+	cn0411_dev->ad7124_dev.regs[AD7124_IOCon1].value =
+		(((ch_gain + 1) << 20)
+		 | (AD7124_8_IO_CTRL1_REG_GPIO_CTRL3
+		    | AD7124_8_IO_CTRL1_REG_GPIO_CTRL2
+		    | AD7124_8_IO_CTRL1_REG_GPIO_CTRL1)
+		 | AD7124_IO_CTRL1_REG_IOUT0(3)
+		 | AD7124_IO_CTRL1_REG_IOUT_CH0(0));
+	ret = AD7124_WriteRegister(&cn0411_dev->ad7124_dev,
+				   cn0411_dev->ad7124_dev.regs[AD7124_IOCon1]);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC IO Control 2 configuration function
+ *
+ * @param cn0411_dev - The device structure.
+ * @return 0 in case of success, negative error code otherwise.
+**/
+int32_t CN0411_ADC_set_io2 (cn0411_device *cn0411_dev)
+{
+	int32_t ret;
+
+	/* All VBias off */
+	cn0411_dev->ad7124_dev.regs[AD7124_IOCon2].value = 0x0000;
+	ret = AD7124_WriteRegister(&cn0411_dev->ad7124_dev,
+				   cn0411_dev->ad7124_dev.regs[AD7124_IOCon2]);
+
+	return ret;
+}
+
+/**
+ * CN0411 ADC conversion mode initialization
+ *
+ * @param cn0411_dev - The device structure.
+ * @param conv_mod - set conversion mode
+ * @return 0 in case of success, negative error code otherwise.
+*/
+int32_t CN0411_ADC_conv_init(cn0411_device *cn0411_dev, uint8_t conv_mod)
+{
+	int32_t ret;
+
+	switch(conv_mod) {
+	case ADC_SINGLE_CONV:
+		ret = CN0411_ADC_ch0 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch3 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch4 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_ch5 (cn0411_dev, ADC_CH_DISABLE);
+		ret |= CN0411_ADC_operation_mode(cn0411_dev, IDLE_MODE);
+		break;
+	case ADC_CONTINUOUS_CONV:
+		ret = CN0411_ADC_ch0 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch1 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch2 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch3 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch4 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_ch5 (cn0411_dev, ADC_CH_ENABLE);
+		ret |= CN0411_ADC_operation_mode(cn0411_dev, CONTINUOUS_CONV);
+		break;
+	default:
+		ret = -1;
```

Comment:
you can return directly for the default clause: `return FAILURE`; and remove the `break`